wave_kinematics/include/wave/kinematics/numerical_test_functors.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef __WAVE_NUMERICAL_TEST_FUNCTORS_HPP__
+#define __WAVE_NUMERICAL_TEST_FUNCTORS_HPP__
+
+#include <iostream>
+#include <functional>
+
+#include "wave/kinematics/rotation.hpp"
+#include "wave/utils/utils.hpp"
+#include <kindr/Core>
+
+namespace wave {
+
+// The Functors are used for computing finite difference Jacobians
+// of rotation expressions.
+
+class RotateAndJacobianJpointFunctor {
+public:
+  wave::Rotation R;
+  wave::Vec3 P;
+  RotateAndJacobianJpointFunctor(wave::Rotation input_rotation) {
+    this->R = input_rotation;
+  }
+
+  int operator()(wave::Vec3 input_point, wave::Vec3 &output_point) {
+    output_point = R.rotate(input_point);
+  }
+};
+
```

Comment:
Indent spacing does not conform to current 4 space indent, consider running the clang-formatter commands I gave you.